src/cirrus/deploy_plugins.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env python
+"""
+_deploy_plugins_
+
+Plugin helpers to talk to various deployment platforms
+
+"""
+from cirrus.logger import get_logger
+from cirrus.configuration import load_configuration
+
+
+LOGGER = get_logger()
+_PLUGINS = {}
+
+
+def bootstrap_parser(subparser):
+    """
+    _bootstrap_parser_
+
+    Util to loop through registered plugin instances
+    and call their build_parser hooks to add them to
+    the CLI suite
+
+    """
+    for plugin, instance in _PLUGINS.iteritems():
+        deployer_command = subparser.add_parser(plugin)
+        instance.build_parser(deployer_command)
+
+
+def get_plugin(plugin_name):
+    """
+    _get_plugin_
+
+    Helper to get the plugin instance based on the name
+    """
+    return _PLUGINS.get(plugin_name)
+
+
+class DeployerRegistry(type):
+    """
+    _DeployerRegistry_
+
+    metaclass to register plugin subclasses with plugins registry
+
+    """
+    def __init__(cls, name, bases, dct):
+        pname = getattr(cls, 'plugin_name')
```

Comment:
.plugin_name is a required attribute, correct?

So it might be better to just do something like:

assert cls.plugin_name is not None

That way, it blows up on someone quickly. (Could also wrap in try - except and catch AttributeError and AssertionError, with an informative message.)